extensions/src/main/java/org/pentaho/platform/plugin/action/javascript/JavascriptRule.java

```diff
@@ -266,6 +267,7 @@ protected Object executeScript( final ScriptableObject scriptable, final Scripta
         results.setResultSet( (IPentahoResultSet) inputValue );
         wrapper = Context.javaToJS( inputValue, results );
       } else {
+        inputValue = Encode.forHtml( Context.toString( inputValue ) );
```

Comment:
I think encoding for html could cause issues here. Suggest killing the encode here as I couldn't demonstrate any ability to subvert the Javascript system based on how we bind the variables. Suggest doing the encode in the success message and calling it a day.